docusaurus/docs/iOS/client/state-layer/state-layer-overview.md

```diff
@@ -0,0 +1,229 @@
+// fetch the first page of channels from the Stream API
+try await channelList.get()
+// fetch more channels
+try await channelList.loadMoreChannels()
+// access all the fetched channels
+channels = channelList.state.channels
+```
+
+Since `ChannelListState` is observable, we can observe and react to channel list changes.
+
+```swift
+let cancellable = channelList.state
+    .$channels
+    .sink { channels in
+        // do something with channels
+    }
+```
+
+### Chat
+
+`Chat` and its `ChatState` represent the state of a channel. In addition, `Chat` has a method for retrieving an observable `MessageState` type which represents a single message and its observable state like reactions and replies.
+
+```swift
+let channelId = ChannelId(type: .messaging, id: "general")
+let chat = chatClient.makeChat(for: channelId)
+// local state of the channel (state from previous sessions)
+var messages = chat.state.messages
+var members = chat.state.members
+// fetch the latest state from the Stream API and subscribe to server-side events
+try await chat.get(watch: true)
+// access all the fetched messages and members
+messages = chat.state.messages
+members = chat.state.members
+// fetch more messages
+try await chat.loadOlderMessages()
+// access all the fetched messages
+messages = chat.state.messages
+```
+
+:::tip Call get with watch true once per app lifetime
+
+The get method fetches the latest state from the Stream API and if we set watch to true, server-side events will keep your local state up to state.
+
+:::
+
+If we would like to observe a single message's state we can use the `MessageState` observable object.
+
+```swift
+let messageState = try await chat.messageState(for: messageId)
+try await chat.loadMoreReactions(for: messageId)
+// access all the fetched reactions
+let reactions = messageState.reactions 
+```
+
+### ConnectedUser
+
+`ConnectedUser` and its `ConnectedUserState` represent the currently logged in user. We can set up push notifications and set states, like muting a user in every channel. Another common use-case is marking all the channels as read.
+
+```swift
+var connectedUser = try await chatClient.connectUser(
+    userInfo: UserInfo(id: "<# Your User ID Here #>"),
+    token: "<# Your User Token Here#>"
+)
+// or while being logged in
+connectedUser = try chatClient.makeConnectedUser()
+// register for push notifications
+try await connectedUser.addDevice(
+    .apn(
+        token: apnTokenData, 
+        providerName: "<# Your Stream's Push Configuration Name>"
+    )
+)
+try await connectedUser.markAllChannelsRead()
+```
+
+### MemberList
+
+`MemberList` and `MemberListState` represent channel members for the specified query. The query supports a wide variety of filters and sorting options. While `MemberList` provides advanced querying capabilities, it is not required if we would like to load all the channel members. For that purpose, `Chat` has built-in paginated loading methods for all the channel members.
```

Comment:
I guess this is the same thing we have on Reactions 🤔 This requires some discussion, but could this be a bit confusing? Having 2 ways to load members, and 2 ways of loading reactions. Can't we have the simple and the more customizable version in 1 API only. For example, if you don't provide a filter or query, it just loads all reaction/members.